### PR TITLE
Fix the behaviour of deletions in trinary_input_char.

### PR DESCRIPTION
**Pull request by conte91**
After inserting the first letter into a trinary_input_char,
the following letters would always require backspace to be pressed
twice in order to be effective, even in the case in which one was
in the main screen (i.e. not selecting a sub-alphabet).
This was due to the code setting the in_progress variable to true
even when a letter had just been selected.

### CODE REVIEW

File: src/ui/components/trinary_input_char.c [hunk 1]
```diff
@@ -61,20 +61,39 @@ typedef struct {
 } data_t;
 
 static void _set_alphabet(component_t* component, const char* alphabet)
+{
+    data_t* data = (data_t*)component->data;
```
Comment (benma) [inline src/ui/components/trinary_input_char.c]: remove this function? it is too small to be worth it ;)

File: src/ui/components/trinary_input_char.c [hunk 2]
```diff
@@ -61,20 +61,39 @@ typedef struct {
...
     }
     if (len == 1) {
-        // Reset elements so that the previous letters don't slide back on a new alphabet.
+        /*
+         * Reset elements so that the previous letters don't slide back
+         * on a new alphabet.
+         */
         memset(data->elements, 0, sizeof(data->elements));
         data->character_chosen_cb(component, alphabet[0]);
         data->in_progress = false;
```
Comment (benma) [inline src/ui/components/trinary_input_char.c]: > This was due to the code setting the in_progress variable to true
> even when a letter had just been selected.

the letter is selected here, so in_progress=false. where was it set to true?

**Comment by benma** [comment]
Oh yeah and please prefix the commit title with the context :nerd_face:

File: src/ui/components/trinary_input_char.c (hunk 1 above)
Comment (conte91) [reply src/ui/components/trinary_input_char.c]: Yeah will do.

**Pull request merged**